pkg/api/types.go

```diff
@@ -1712,6 +1714,103 @@ type SecretList struct {
 	Items []Secret `json:"items"`
 }
 
+// SecurityContext holds security configuration that will be applied to a container.  If a security context
+// is set on the container spec then it must comply with any constraints defined in the SecurityConstraints context
+// it is running in.  If a security context is not supplied and the pod is running under a SecurityConstraints context
+// then a default SecurityContext may be applied.
+type SecurityContext struct {
+	// Capabilities are the capabilities to add/drop when running the container
+	// TODO: will need to refactor this from the container spec to here
+	Capabilities Capabilities
+
+	// Run the container in privileged mode
+	// TODO: will need to refactor this from the container spec to here
+	Privileged bool
```

Comment:
I am NOT a security person, so I am just speaking from aesthetics.  I like
the model that security settings enable other flags but do not replace
other flags.  Especially when they go 1:N like HostPorts.

It's not clear what I would write, as a user, to enable HostPorts in this
externalized mode.

On Tue, Apr 7, 2015 at 4:02 PM, Clayton Coleman <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6287#discussion_r27929692>
> :
>
> > @@ -1712,6 +1714,103 @@ type SecretList struct {
> >  	Items []Secret `json:"items"`
> >  }
> >
> > +// SecurityContext holds security configuration that will be applied to a container.  If a security context
> > +// is set on the container spec then it must comply with any constraints defined in the SecurityConstraints context
> > +// it is running in.  If a security context is not supplied and the pod is running under a SecurityConstraints context
> > +// then a default SecurityContext may be applied.
> > +type SecurityContext struct {
> > +	// Capabilities are the capabilities to add/drop when running the container
> > +	// TODO: will need to refactor this from the container spec to here
> > +	Capabilities Capabilities
> > +
> > +	// Run the container in privileged mode
> > +	// TODO: will need to refactor this from the container spec to here
> > +	Privileged bool
>
> Agree Host Ports is weird. Privileged though overlaps with capabilities,
> which overlaps with namespaces. It's possible that we could not have a
> grouping for many of these - I liked the idea of the serious security
> problems all being in one spot. But I'm not married to it.
> … <#14c961ebe3c6ec68_>
> ----- Original Message -----
>  > @@ -1712,6 +1714,103 @@ type SecretList struct { > Items []Secret
> `json:"items"` > } > > +// SecurityContext holds security configuration
> that will be applied to a > container. If a security context > +// is set
> on the container spec then it must comply with any constraints > defined in
> the SecurityConstraints context > +// it is running in. If a security
> context is not supplied and the pod is > running under a
> SecurityConstraints context > +// then a default SecurityContext may be
> applied. > +type SecurityContext struct { > + // Capabilities are the
> capabilities to add/drop when running the > container > + // TODO: will
> need to refactor this from the container spec to here > + Capabilities
> Capabilities > + > + // Run the container in privileged mode > + // TODO:
> will need to refactor this from the container spec to here > + Privileged
> bool Moving this out of the container object starts to feel weird when you
> apply it to host networking and even weirder when you apply it to host
> ports. Is this the right pattern? --- Reply to this email directly or view
> it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/6287/files#r27928586
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6287/files#r27929692>
> .
>